internal/entimport/import.go

```diff
@@ -203,3 +205,157 @@ func isJoinTable(table *schema.Table) bool {
 	}
 	return true
 }
+
+func typeName(tableName string) string {
+	return inflect.Camelize(inflect.Singularize(tableName))
+}
+
+func tableName(typeName string) string {
+	return inflect.Underscore(inflect.Pluralize(typeName))
+}
+
+func resolvePrimaryKey(importer SchemaImporter, table *schema.Table) (f ent.Field, err error) {
+	if table.PrimaryKey == nil || len(table.PrimaryKey.Parts) != 1 {
+		return nil, fmt.Errorf("entimport: invalid primary key - single part key must be present")
+	}
+	if f, err = importer.field(table.PrimaryKey.Parts[0].C); err != nil {
+		return nil, err
+	}
+	if f.Descriptor().Name != "id" {
+		f.Descriptor().StorageKey = f.Descriptor().Name
+		f.Descriptor().Name = "id"
+	}
+	return f, nil
+}
+
+func upsertNode(i SchemaImporter, table *schema.Table) (*schemast.UpsertSchema, error) {
+	upsert := &schemast.UpsertSchema{
+		Name: typeName(table.Name),
+	}
+	fields := make(map[string]ent.Field, len(upsert.Fields))
+	for _, f := range upsert.Fields {
+		fields[f.Descriptor().Name] = f
+	}
+	pk, err := resolvePrimaryKey(i, table)
+	if err != nil {
+		return nil, err
+	}
+	if _, ok := fields[pk.Descriptor().Name]; !ok {
+		fields[pk.Descriptor().Name] = pk
+		upsert.Fields = append(upsert.Fields, pk)
+	}
+	for _, column := range table.Columns {
+		if column.Name == table.PrimaryKey.Parts[0].C.Name {
```

Comment:
Possible panic. I'd add a check, even when you check it "indirectly" in `resolvePrimaryKey`.